### PR TITLE
Update minimum supported pyamg

### PR DESCRIPTION
**Pull request by jarrodmillman**
## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpydoc/blob/main/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
...
```

### CODE REVIEW

**Comment by mkcor** [comment]
Relates to #7576

**Review by mkcor — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -93,7 +93,7 @@ optional = [
     'dask[array]>=2021.1.0,!=2024.8.0',
     'matplotlib>=3.7',
     'pooch>=1.6.0',
-    'pyamg',
+    'pyamg>=5.2',
```
Comment (hmaarrfk) [inline pyproject.toml:96]: 5.2 is SUPER new.

```
$ nep29 pyamg
| version |    date    |
| :-----: | :--------: |
|  5.2.1  | 2024-06-27 |
|  5.1.0  | 2024-03-19 |
|  5.0.0  | 2023-04-17 |
```

Comment (hmaarrfk) [reply pyproject.toml:96]: I generally don't have aproblem with this, but I just wanted to point it out.

Comment (stefanv) [reply pyproject.toml:96]: I believe we need this version because of sparse dtype issues. As an optional, speed-enhancing dependency, I think it's OK.

**Review by stefanv — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (hmaarrfk) [reply pyproject.toml:96]: Ok

**Review by hmaarrfk — APPROVED** [review]

**Pull request merged**